gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageStatistics.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.hadoop.gcsio;
+
+import static com.google.cloud.hadoop.gcsio.StatisticTypeEnum.TYPE_COUNTER;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Maps;
+import java.util.EnumSet;
+
+/** Statistics which are collected in GCS Client Side. */
+public enum GoogleCloudStorageStatistics {
+
+  /** Client-side Status Code statistics */
+  GCS_REQUEST_COUNT(
+      "gcs_total_request_count", "Counts the total number of gcs requests made", TYPE_COUNTER),
+
+  GCS_EXCEPTION_COUNT(
```

Comment:
Should have move the exception count out of GCS? i.e. let us have an exception count for entire connector? I agree having a bifurcation for GCS related exceptions will be useful. We would like to know connector level exception count as well.

Reply:
So, would it be better to bifurcate now into EXCEPTION_COUNT and GCS_EXCEPTION_COUNT ? Other thing we can do is to go with EXCEPTION_COUNT and later in future define another counter specific for GCS.